src/main.c

```diff
@@ -29,15 +29,41 @@ int main(int argc, char *argv[])
 {
   LOG("STARTING SAT SOLVER", 3);
 
-  if (!check_args(argc, argv)) { return 0; }
-  
-  // TODO: malloc structs
+  FILE* fp = check_args(argc, argv);
+
+  if(!fp) 
+  { 
+  	LOG("Error while opening/reading file.", 3); 
+  	printf("ERROR\n");
+  	input_free(fp, NULL, NULL);
+  	return 0; 
+  }
   
-  if (!input_parser (NULL, NULL, NULL)) { printf("ERROR\n"); return 0; } // TODO fix args
-  if (!solve        (NULL, NULL      )) { printf("ERROR\n"); return 0; } // TODO fix args
-  if (!input_free   (NULL, NULL, NULL)) { printf("ERROR\n"); return 0; } // TODO fix args
+  UNMOLESTED_INPUT *unin = malloc(sizeof(UNMOLESTED_INPUT));
+  MOLESTED_INPUT *in = malloc(sizeof(MOLESTED_INPUT));
```

Comment:
If we malloc them in main, shouldn't we free them in main too?
(e.g. do we really need the input_free call at all?)